src/v/cloud_topics/reconciler/range_batch_consumer.cc

```diff
@@ -14,51 +14,54 @@
 
 namespace cloud_topics::reconciler {
 
+range_batch_consumer::range_batch_consumer(
+  l1::object_builder* builder, model::ntp ntp)
+  : _builder(builder)
+  , _ntp(std::move(ntp)) {}
+
 ss::future<ss::stop_iteration>
 range_batch_consumer::operator()(model::record_batch batch) {
     if (!_base_offset.has_value()) {
         _base_offset = model::offset_cast(batch.base_offset());
+        co_await _builder->start_partition(_ntp);
     }
+
     // NOTE: we only have data batches here so it's safe
-    // to use timestamps without checking the batch type
-    if (_range.info.base_timestamp == model::timestamp{}) {
-        _range.info.base_timestamp = batch.header().first_timestamp;
+    // to use timestamps without checking the batch type.
+    if (_metadata.base_timestamp == model::timestamp{}) {
+        _metadata.base_timestamp = batch.header().first_timestamp;
     } else {
-        _range.info.base_timestamp = std::min(
-          batch.header().first_timestamp, _range.info.base_timestamp);
+        _metadata.base_timestamp = std::min(
+          batch.header().first_timestamp, _metadata.base_timestamp);
     }
-    _range.info.last_timestamp = std::max(
-      batch.header().max_timestamp, _range.info.last_timestamp);
-    _range.info.last_offset = model::offset_cast(batch.last_offset());
+    _metadata.last_timestamp = std::max(
+      batch.header().max_timestamp, _metadata.last_timestamp);
+    _metadata.last_offset = model::offset_cast(batch.last_offset());
 
     bool add_term = false;
-    if (_range.info.terms.empty()) {
-        // Always add term of the first batch in the sequence. We don't really
-        // know if it's the first batch in the term so some filtering will be
-        // needed later.
+    if (_metadata.terms.empty()) {
```

Comment:
nit: not from this PR, but IMO this is easier to follow as:

```
if (_metadata.terms.empty() || batch.term > _metadata.terms.rbegin()->first) {
    _metadata.terms.insert(batch.term, batch.base_offset());
}
```

The original `!=` here instead of `>` is kinda confusing -- i'm guessing it's to be defensive, but it feels like an edge case that's unlikely to be useful since terms should always increase